readme.txt
--------------------------------------------------------------------------------
                  Name : Sai Midhil Chowdary Kari
                    Email : midhilkari@gmail.com
--------------------------------------------------------------------------------


--------------------------------------------------------------------------------
APPLICATION NAME  :       LEARN IT
Developer         :       Sai Midhil Chowdary Kari
--------------------------------------------------------------------------------

OVERVIEW :

LEARN IT is a web app targeting age group of 4 - 9 years kids. This game tries
to train up the player's mind by taking them through different levels of
constructively increasing difficulty. This game consists of learning modules for
NUMBERS & WORDS. Each module consists of seven levels currently and the player
can get access to any level. The difficulty in questions increases with increase
in levels. The colour mix used in the game is carefully selected keeping in mind
the targeted age group for the game resulting in less strain on their eyes. This
application pre-assumes basic knowledge of mathematics and general english words.

--------------------------------------------------------------------------------
--------------------------------------------------------------------------------


                        DIFFERENT GAME PAGES
--------------------------------------------------------------------------------

landingPage : This is the page user views when the game is loaded. The title and
the start button would be sitting in this page.

levelSelectionPage : This is the page user retrieves into when he clicks on the
"Let's Learn" button i.e, when the player starts playing game. Here in this page
the player would get option to choose from two options of learning  WORDS or
NUMBERS. Subjective to the choice of the player's selection of learning words or
numbers would be retrieved to level selection page.

levelContent : In each sections of learning, respective level related questions
would be displayed on the screen to be answered by the player. If the entered
answer is correct player would encounter the next question or else would be
getting reply of "Wrong Answer". If all the question's in a particular level are
cleared then player would be moved to next level and would face a bit more
critical questions.

--------------------------------------------------------------------------------


                            MAJOR FEATURES
--------------------------------------------------------------------------------

- Currently there are seven levels in each of the learning modules.
- A speaker button would be present on top right corner through out the game in
  the same position. It can be toggled to turn music ON or OFF.
- Number of levels can be configured from the "config.js" file directly. Change
  in number of levels in this file would reflect directly in the game. So any
  person with no knowledge about the code can configure number of levels to be
  present in the game.
- The data used in the game is loaded from a separate JS file so that the
  questions & answers data can be directly modified from the data
  file(question.js).
- The content on every button, heading, paragraph is loaded dynamically from the
  "tittlesData.js" file. SO to change any content in the page can be done
  directly form here.
- Applauding music is played after every level cleared to cheer up the user.


--------------------------------------------------------------------------------


                    TECHNICAL REFERENCES FOR DEVELOPERS
--------------------------------------------------------------------------------

- config.js : Number of levels to be present in the game can be configured from
  here.
- questions.js : All the questions & answers data is present in this file.
- app.js : All the game level logical code is present in this file.
- gameElements.js : All the logical code of the game pages content is present in
  this file. All  the logic for to get level respective content is developed
  here.
- ui.js : All the elements used in the game are generated here and are re-used
  in various parts of the code when ever required.
- titlesData.js : All the content displayed on different elements of the game
  are populated here.
- licenses.txt : All the attributions for all the content used in the game are
  present here.
- settings.js : The logic for toggling music is present in this file.

--------------------------------------------------------------------------------


                              USEFUL FUNCTIONS
--------------------------------------------------------------------------------

- loadLandingPage(): Loads the landing page content of the game. The title and
  the start button will be created in this function.
  This in turn calls createLandingPageContent() function to create aforesaid
  content.

- loadLevelSelectionPage(): Loads level selection page of learning words &
  numbers modules.
  This in turn calls createLevelSelectionPage() function to create aforesaid
  content.

- loadNumbersPage(): Loads numbers page with the all level options to be played
  by the user.
  This in turn calls numbers() function to create aforesaid
  content.

- loadWordsPage(): Loads words page with the all level options to be played
  by the user.
  This in turn calls words() function to create aforesaid
  content.

- loadNumbersLevelPage(): Loads respective level content in the numbers module.
  This in turn calls createNumbersLevelContent() function to create aforesaid
  content.

- loadWordsLevelPage(): Loads respective level content in the numbers module.
  This in turn calls createWordsLevelContent() function to create aforesaid
  content.

- updateNumberQuestion(): This function updates the questions after answering
  each question. It ensures to not repeat same questions and to levlUP user
  after answering all the questions in particular level.

- numbersLevelClearedAnimation(): This function loads the animation that is
  generated after clearing each level.

- updateWordQuestion(): This function updates the questions after answering
  each question. It ensures to not repeat same questions and to levlUP user
  after answering all the questions in particular level.

- wordsLevelClearedAnimation(): This function loads the animation that is
  generated after clearing each level.

--------------------------------------------------------------------------------
ui.js : It contains all the functions that creates elements that are used all
through the game.
- createDiv(): Creates Division.
- createPara(): Creates paragraph.
- createImage(): Creates Image.
- createTextInput(): Creates text input box.
- createButton(): Creates button.
- createBreakLine(): Creates break line.
--------------------------------------------------------------------------------

                                  NOTES
- Keeping in time constraints in mind, a very limited data & levels are included
  in this game. But, further levels & questions can be easily configured from
  the config.js & questionsData.js files by anyone with no programming knowledge.
- All the colours used in this game are developed keeping target age group
  audience in mind!!

  --------------------------------------------------------------------------------


changelog.txt
--------------------------------------------------------------------------------
                                CHANGE LOG
--------------------------------------------------------------------------------

Week 4, Wednesday, 27/March
Finished
- Initial data structure for questions & answers is created.
- Styled application according to the guidelines mentioned in the project plan.
  i.e, Used black colour background keeping target users for this game in mind
  to reduce stress on their eyes while using this application.

Still working on
- Creating elements to be used in the application.

Milestone progress
- All the required JavaScript files are linked in index file.

--------------------------------------------------------------------------------
Week 4, Thursday, 28/March
Finished
- Data structure of questions & answers is modified according to the
  requirements.

Still working on
- Collecting required data for questions & answers.

Milestone progress
- Most of the initial required for testing the application is acquired.

--------------------------------------------------------------------------------
Week 5, Wednesday, 3/April
Finished
- Categorising JavaScript files according to granularity levels.
- Defined functions to load landing page & loadLevelSelectionPage.

Still working on
- Levels page content & layout of each learning module.

Milestone progress
- Target to get basic layout of the application is 50% achieved.

--------------------------------------------------------------------------------
Week 6, Wednesday, 10/April
Finished
- Basic layout of the application.
- Defined functions to create levels dynamically according to configured levels
  in the config,js file.

Still working on
- Adding level wise content to the levels & randomly selecting  questions.

Milestone progress
- Basic layout is created for the application.

--------------------------------------------------------------------------------
Week 6,Sunday, 14/April
Finished
- Defined createNumbersLevelContent() & createWordsLevelContent() functions to
  generate respective learning modules content.

Still working on
- Randomly getting questions from the data.

Milestone progress
- Able to load data to the level contents page.

--------------------------------------------------------------------------------


Week 7,Thursday, 18/April
Finished
- Defined shuffle() function to randomly select elements from the questions
  array.
- Added more questions data.

Still working on
- Handling bugs to load data correctly.

Milestone progress
- Apart from bug of words questions & numbers questions getting mixed up, able
  to get data randomly.

--------------------------------------------------------------------------------
Week 7, Friday, 19/April
Finished
- Downloaded good vector images content for where ever required.

Still Working on
- Resolving bug on mixing of words & numbers questions.
- Adding background music to the application.

Milestone progress
- No much progress since the last log except that more vector images are
  downloaded.

--------------------------------------------------------------------------------
Week 8, Friday, 3/May
Finished
- Resolved bug of random mixing of numbers & words data.

Still working on
- Consistent layout for the application.

Milestone progress
- Almost the basic structure of the application is ready & the game is in
  playable situation except that pf having any background music.

--------------------------------------------------------------------------------
Week 8, Sunday, 5/May
Finished
- Added logic for music toggling.

Still working on
- Styling for different layout of the game.

Milestone progress
- Apart from very few disturbances in the layout the game is almost completed.

--------------------------------------------------------------------------------
Week 9, Monday, 6/May
Finished
- Achieved consistent layout for all the mode mentioned in the marking ruberics.

Still working on
- Writing well explained comments.

Milestone progress
- 90% of all the required tasks are done for the project ZIP file to be
  uploaded!!

--------------------------------------------------------------------------------
Week 10, Monday, 13/May
Finished
- Completed comments for each and every function in my code.
- Renamed some functions to be easily understandable to anyone who reads my code.

Still working on
- Making Demonstration video.

--------------------------------------------------------------------------------
Week 10,Tuesday, 14/May
Finished
- Sample demonstration video is made.

Still working on
- Generating demonstration video.

--------------------------------------------------------------------------------
Week 10,Wednesday, 15/May
Finished
- Final Demonstration video is made.

Still working on
- Editing of demonstration video.

--------------------------------------------------------------------------------
Week 10,Thursday, 16/May
Finished
- Editing & Uploading Demonstration video.
- 100% of the work is done the ZIP file is ready to be uploaded.

--------------------------------------------------------------------------------


js/ui.js
/*
  ui.js: This file consists of all the elements used in our application
  Author: Sai Midhil Chowdary Kari
  Email: midhilkari@gmail.com
*/

/*
  This function creates heading 1
  Parameters:
  content : Content to be displayed as heading 1
  Example:
  createHeading1("Heading")
  creates : <h1>Heading</h1>
*/
function createHeading1(content) {
  var element = document.createElement("H1");
  if(content != undefined) {
    var text = document.createTextNode(content);
    element.appendChild(text);
  }
  return element;
}

/*
  This function creates division
  Parameters:
  content : Content to be displayed in the division
  Example:
  createDiv("Division")
  creates : <div>Division</div>
*/
function createDiv(content) {
  var element = document.createElement("DIV");
  if(content != undefined) {
    var text = document.createTextNode(content);
    element.appendChild(text);
  }
  return element;
}

/*
  This function creates Button
  Parameters:
  content : Content to be displayed on the button
  Example:
  createButton("Button")
  creates : <button>Button</button>
*/
function createButton(content) {
  var element = document.createElement("BUTTON");
  if(content != undefined) {
    var text = document.createTextNode(content);
    element.appendChild(text);
  }
  return element;
}

/*
  This function creates paragraph
  Parameters:
  content : Content to be displayed in the paragraph
  Example:
  createPara("Paragraph")
  creates : <p>Paragraph</p>
*/
function createPara(content) {
  var element = document.createElement("p");
  if(content != undefined) {
    var text = document.createTextNode(content);
    element.appendChild(text);
  }
  return element;
}

/*
  This function creates text input element and returns text
*/
function createTextInput() {
  var element = document.createElement("INPUT");
  element.setAttribute("type","text");
  return element;
}

/*
  This function creates break line
  This is used to create space between elements
*/
function createBreakLine(){
  return document.createElement('br');
}

/*
  This function creates image element
  Parameters:
  content : Content to be displayed when the image is not loaded
  imageUrl : link to the image
  Example:
  createImg("Image not loaded","sampleimage.svg")
  creates : <img alt="Image not loaded" src="sampleimage.svg">
*/
function createImage(content, imageUrl) {
  var element = document.createElement("IMG");
  element.setAttribute("alt",content);
  element.setAttribute("src",imageUrl);
  return element;
}

/*
  This function creates span element
  Parameters:
  content : Content to be displayed in the span
  Example:
  createSpan("This is span")
  creates : <span>This is span</span>
*/
function createSpan(content) {
  var element = document.createElement("SPAN");
  if(content != undefined) {
    var text = document.createTextNode(content);
    element.appendChild(text);
  }
  return element;
}


js/app.js
/*
  app.js:This file loades all application related logic
  Author: Sai Midhil Chowdary Kari
  Email: midhilkari@gmail.com
*/

//loading loadLandingPage function on load of application
window.onload = loadLandingPage;

/*
  loadLandingPage(): Loads landing page content
  returns content in landing page of the game
*/
function loadLandingPage(){
  console.log("loading landing page content");
  loadSounds();
  var landingPageContent = createLandingPageContent();
  document.body.innerHTML = "";
  document.body.appendChild(landingPageContent);
}

/*
  loadLevelSelectionPage(): Loads level selection page of learning words &
  numbers
  returns lavel section page of learning words & numbers
*/
function loadLevelSelectionPage(){
  console.log("Loading level selection page loaded");
  var levelSelectionPage  = createLevelSelectionPage();
  document.body.innerHTML = "";
  document.body.appendChild(levelSelectionPage);
}

/*
  loadNumbersPage(): Loads numbers learning page
  loads numbers learning page
*/
function loadNumbersPage(){
  console.log("Number's learning section selected");
  var n = numbersPage();
  document.body.innerHTML = "";
  document.body.appendChild(n);
}

/*
  loadNumbersPage(): Loads words learning page
  loads words learning page
*/
function loadWordsPage(){
  console.log("Word's learning section selected");
  var w = wordsPage();
  document.body.innerHTML = "";
  document.body.appendChild(w);
}

//setting no.of levels intially as 0
window.level = 0;
/*
  loadNumbersLevelPage(): Loads numbers number of levls section
*/
function loadNumbersLevelPage(level) {
  console.log("Loading number's level content");
  var numbersLevelContent = createNumbersLevelContent(level);
  document.body.innerHTML = "";
  document.body.appendChild(numbersLevelContent);
}

/*
  loadWordsLevelPage(): Loads words number of levls section
*/
function loadWordsLevelPage(level) {
  console.log("Loading word's level content");
  var wordsLevelContent = createWordsLevelContent(level);
  document.body.innerHTML = "";
  document.body.appendChild(wordsLevelContent);
}


js/settings.js
/*
  settings.js: This file consists of functions that handle music setting in the
  application
  Author: Sai Midhil Chowdary Kari
  Email: midhilkari@gmail.com
*/

/*
  loadSounds(): This function loades music tracks used in the application
*/
function loadSounds() {
  //Loading the music into global variable sounds
  window.sounds = {
    "game": new Audio("sounds/backgroundMusic.mp3"),
    "levelUp": new Audio("sounds/levelUp.mp3"),
  }
  //The speaker image to be shown when speaker is on
  window.musicOn = createImage('Image Not Found','images/speakerOn.svg');
  //Event handler to toggle music on click of the icon
  musicOn.onclick = toggleMusic;
  //The speaker image to be shown when speaker is off
  window.musicOff = createImage('Image Not Found','images/speakerOff.svg');
  //Event handler to toggle music on click of the icon
  musicOff.onclick = toggleMusic;
}

window.speakerImage = 'speakerOff';
/*
  toggleMusic(): This function turns music ON when the speakerOn icon is
  clicked & turns music OFF when speakerOff icon is clicked
*/
function toggleMusic(){
  //If speaker is of condition then on click make it on and start music
  if(speakerImage === 'speakerOff'){
    //Change speakerImage to speakerOn
    speakerImage = 'speakerOn';
    //Start the game music in loop
    sounds.game.loop = true;
    sounds.game.play();
    //Empty the speaker div
    document.getElementById('speaker').innerHTML = '';
    //Append music on icon
    var speakerOnDiv = createDiv();
    speakerOnDiv.classList.add('speakerCOntainer');
    document.getElementById('speaker').appendChild(musicOn);
  } else {
    //Change speakerImage to speakerOff
    speakerImage = 'speakerOff';
    //If music is turned on stop it
    //Turn off level up music if its on
    if(sounds.levelUp != undefined){
      sounds.levelUp.pause();
      sounds.levelUp.currentTime = 0.0;
    }
    //Turn off game music if its on
    if(sounds.game != undefined){
      sounds.game.pause();
      sounds.game.currentTime = 0.0;
    }
    //Clear the content in speaker container
    document.getElementById('speaker').innerHTML = '';
    //Append music off icon
    var speakerOffDiv = createDiv();
    speakerOffDiv.classList.add('speakerCOntainer');
    document.getElementById('speaker').appendChild(musicOff);

  }
}

/*
  createSpeakerIcon(): This function creates speakerON & speakerOff icons and
  turns music tracks playing when the icons were clicked
*/
function createSpeakerIcon() {
  //Creates a division to hold the speaker
  var speakerDiv = createDiv();
  //Setting an ID for easy toggling of image on music change
  speakerDiv.setAttribute('id','speaker');
  //Adding the speaker image as per the music is on or off currently
  var speakerImg = createImage('Image Not Found','images/'+speakerImage+'.svg');
  speakerImg.classList.add('headerIcon');
  //Adding event handler to toggle music when clicking on the speaker icon
  speakerImg.onclick = toggleMusic;
  //Appending the speaker image to the speaker container
  speakerDiv.appendChild(speakerImg);
  //returning the element with speaker content
  return speakerDiv;
}


js/utilityFunctions.js
/*
  utilityFunction.js: This file contains all the logical functions used in the
  application
*/

/*
  shuffle(array): This function shuffle's the elements in the array. This is
  used to shuffle questions avoiding them to populate them in the same order
  the level is selected
  URL: https://gomakethings.com/how-to-shuffle-an-array-with-vanilla-js/
*/
function shuffle(array) {

	var currentIndex = array.length;
	var temporaryValue, randomIndex;

	// While there remain elements to shuffle
	while (0 !== currentIndex) {
		// Pick a remaining element...
		randomIndex = Math.floor(Math.random() * currentIndex);
		currentIndex -= 1;
		// And swap it with the current element.
		temporaryValue = array[currentIndex];
		array[currentIndex] = array[randomIndex];
		array[randomIndex] = temporaryValue;
	}
  
	return array;

};


data/questions.js
/*
  questions.js: This file contains all the data of questions and answers used
  in our application
*/

//Numbers data
window.numbersQuestions = [
  {
    questionId : 1,
    question : "How much is '12 + 8' ? ",
    answer: "20",
    level: 1
  },
  {
    questionId : 2,
    question: "How much is '8 + 12' ?",
    answer: "20",
    level: 1
  },
  {
    questionId : 3,
    question: "How much is '9 + 21' ?",
    answer: "30",
    level: 1
  },
  {
    questionId : 4,
    question: "How much is '15 - 7' ?",
    answer: "8",
    level: 1
  },
  {
    questionId : 5,
    question: "How much is '23 - 13' ?",
    answer: "10",
    level: 1
  },
  {
    questionId : 6,
    question: "How much is '6 + 6' ?",
    answer: "12",
    level: 1
  },
  {
    questionId : 7,
    question: "How much is '2 + 4 + 5' ?",
    answer: "11",
    level: 2
  },
  {
    questionId : 8,
    question: "How much is '5 + 3 + 4' ?",
    answer: "12",
    level: 2
  },
  {
    questionId : 9,
    question: "How much is '7 + 3 + 8' ?",
    answer: "18",
    level: 2
  },
  {
    questionId : 10,
    question: "How much is '1 + 9 + 3' ?",
    answer: "13",
    level: 2
  },
  {
    questionId : 11,
    question: "How much is '6 + 7 + 4' ?",
    answer: "17",
    level: 2
  },
  {
    questionId : 12,
    question: "How much is '5 + 9 + 8' ?",
    answer: "22",
    level: 2
  },
  {
    questionId : 13,
    question: "How much is '2 - 3 + 5' ?",
    answer: "4",
    level: 3
  },
  {
    questionId : 14,
    question: "How much is '1 + 7 - 4' ?",
    answer: "4",
    level: 3
  },
  {
    questionId : 15,
    question: "How much is '4 - 3 + 9' ?",
    answer: "10",
    level: 3
  },
  {
    questionId : 16,
    question: "How much is '5 + 7 - 9' ?",
    answer: "3",
    level: 3
  },
  {
    questionId : 17,
    question: "How much is '7 - 8 + 5' ?",
    answer: "4",
    level: 3
  },
  {
    questionId : 18,
    question: "How much is '8 - 2 + 6' ?",
    answer: "12",
    level: 3
  },
  {
    questionId : 19,
    question: "How much is '2 * 3' ?",
    answer: "6",
    level: 4
  },
  {
    questionId : 20,
    question: "How much is '9 * 2' ?",
    answer: "18",
    level: 4
  },
  {
    questionId : 21,
    question: "How much is '3 * 4' ?",
    answer: "12",
    level: 4
  },
  {
    questionId : 22,
    question: "How much is  '4 * 4' ?",
    answer: "16",
    level: 4
  },
  {
    questionId : 23,
    question: "How much is '5 * 7' ?",
    answer: "35",
    level: 4
  },
  {
    questionId : 24,
    question: "How much is '7 * 3' ?",
    answer: "21",
    level: 4
  },
  {
    questionId : 25,
    question: "How much is '10 / 5' ?",
    answer: "2",
    level: 5
  },
  {
    questionId : 26,
    question: "How much is '49 / 7' ?",
    answer: "7",
    level: 5
  },
  {
    questionId : 27,
    question: "How much is '18 / 2' ?",
    answer: "9",
    level: 5
  },
  {
    questionId : 28,
    question: "How much is '72 / 9' ?",
    answer: "8",
    level: 5
  },
  {
    questionId : 29,
    question: "How much is '16 / 4' ?",
    answer: "4",
    level: 5
  },
  {
    questionId : 30,
    question: "How much is '42 / 6' ?",
    answer: "7",
    level: 5
  },
  {
    questionId : 61,
    question: "How much is '34 + 76' ?",
    answer: "80",
    level: 6
  },
  {
    questionId : 62,
    question: "How much is '20 + 11' ?",
    answer: "31",
    level: 6
  },
  {
    questionId : 63,
    question: "How much is '63 + 18' ?",
    answer: "81",
    level: 6
  },
  {
    questionId : 64,
    question: "How much is '81 + 37' ?",
    answer: "44",
    level: 6
  },
  {
    questionId : 65,
    question: "How much is '74 - 43' ?",
    answer: "31",
    level: 6
  },
  {
    questionId : 66,
    question: "How much is '63 - 12' ?",
    answer: "51",
    level: 6
  },
  {
    questionId : 67,
    question: "How much is '(57 - 32) * 2' ?",
    answer: "50",
    level: 7
  },
  {
    questionId : 68,
    question: "How much is '(12 + 8) * 4' ?",
    answer: "80",
    level: 7
  },
  {
    questionId : 69,
    question: "How much is '(87 - 42) * 3' ?",
    answer: "135",
    level: 7
  },
  {
    questionId : 70,
    question: "How much is '(76 +14) / 3' ?",
    answer: "30",
    level: 7
  },
  {
    questionId : 71,
    question: "How much is '(73 - 57) / 8' ?",
    answer: "2",
    level: 7
  },
  {
    questionId : 72,
    question: "How much is '(43 + 27) / 7' ?",
    answer: "10",
    level: 7
  }
]


//words data
window.wordsQuestions = [
  {
    questionId : 31,
    question : "What's the name of this fruit ?",
    imageUrl: "images/apple.jpeg",
    answer: "apple",
    level: 1
  },
  {
    questionId : 32,
    question: "What's the name of this fruit ?",
    imageUrl: "images/banana.jpeg",
    answer: "banana",
    level: 1
  },
  {
    questionId : 33,
    question: "What's the name of this animal ?",
    imageUrl: "images/cat.jpeg",
    answer: "cat",
    level: 1
  },
  {
    questionId : 34,
    question: "What's the name of this animal ?'",
    imageUrl: "images/dog.jpeg",
    answer: "dog",
    level: 1
  },
  {
    questionId : 35,
    question: "What's the name of this animal ?'",
    imageUrl: "images/elephant.jpeg",
    answer: "elephant",
    level: 1
  },
  {
    questionId : 36,
    question: "What's the name of this object ?",
    imageUrl: "images/fan.jpeg",
    answer: "fan",
    level: 1
  },
  {
    questionId : 37,
    question: "What colour is this?",
    imageUrl: "images/green.jpeg",
    answer: "green",
    level: 2
  },
  {
    questionId : 38,
    question: "What colour is this?",
    imageUrl: "images/blue.jpeg",
    answer: "blue",
    level: 2
  },
  {
    questionId : 39,
    question: "What colour is this?",
    imageUrl: "images/yellow.jpeg",
    answer: "yellow",
    level: 2
  },
  {
    questionId : 40,
    question: "What colour is this?",
    imageUrl: "images/grey.jpeg",
    answer: "grey",
    level: 2
  },
  {
    questionId : 41,
    question: "What colour is this?",
    imageUrl: "images/orange.jpeg",
    answer: "orange",
    level: 2
  },
  {
    questionId : 42,
    question: "What colour is this?",
    imageUrl: "images/red.jpeg",
    answer: "red",
    level: 2
  },
  {
    questionId : 43,
    question: "Which alphabet comes after 'G' ?",
    imageUrl: "",
    answer: "h",
    level: 3
  },
  {
    questionId : 44,
    question: "Which alphabet comes before 'L' ?",
    imageUrl: "",
    answer: "k",
    level: 3
  },
  {
    questionId : 45,
    question: "Which alphabet comes in between 'K' & 'M' ?",
    imageUrl: "",
    answer: "l",
    level: 3
  },
  {
    questionId : 46,
    question: "Which alphabet comes in between 'E' & 'G' ?",
    imageUrl: "",
    answer: "f",
    level: 3
  },
  {
    questionId : 47,
    question: "Which alphabet comes before 'O' ?",
    imageUrl: "",
    answer: "p",
    level: 3
  },
  {
    questionId : 48,
    question: "Which alphabet comes after 'R' ?",
    imageUrl: "",
    answer: "s",
    level: 3
  },
  {
    questionId : 49,
    question: "Complete the following word 'AE __ O __ LA __ N __' !",
    imageUrl: "images/aeroplane.jpeg",
    answer: "aeroplane",
    level: 4
  },
  {
    questionId : 50,
    question: "Complete the following word 'G __ AP __ S' !",
    imageUrl: "images/grapes.jpeg",
    answer: "grapes",
    level: 4
  },
  {
    questionId : 51,
    question: "Complete the following word 'TE __ EVI __ I __ N' !",
    imageUrl: "images/television.jpeg",
    answer: "television",
    level: 4
  },
  {
    questionId : 52,
    question: "Complete the following word 'CH __ CO __ A __ E' !",
    imageUrl: "images/chocolate.jpeg",
    answer: "chocolate",
    level: 4
  },
  {
    questionId : 53,
    question: "Complete the following word 'C __ R' !",
    imageUrl: "images/car.jpeg",
    answer: "car",
    level: 4
  },
  {
    questionId : 54,
    question: "Complete the following word 'T __ A __ N' !",
    imageUrl: "images/train.jpeg",
    answer: "train",
    level: 4
  },
  {
    questionId : 55,
    question: "Complete the sentence 'I ___ a Boy'",
    imageUrl: "",
    answer: "am",
    level: 5
  },
  {
    questionId : 56,
    question: "Complete the sentence, She _____ a Girl",
    imageUrl: "",
    answer: "is",
    level: 5
  },
  {
    questionId : 57,
    question: "Complete the sentence, They _____ playing Cricket",
    imageUrl: "",
    answer: "are",
    level: 5
  },
  {
    questionId : 58,
    question: "Complete the sentence, John _____ three Chocolates!",
    imageUrl: "",
    answer: "had",
    level: 5
  },
  {
    questionId : 59,
    question: "Complete the sentence, He had _____ playing Football since 4PM",
    imageUrl: "",
    answer: "been",
    level: 5
  },
  {
    questionId : 60,
    question: "Complete the sentence, This factory is manufacturing shirts _____ 1970",
    imageUrl: "",
    answer: "since",
    level: 5
  },
  {
    questionId : 73,
    question: "Complete the following word '__ __ O __ S' !",
    imageUrl: "images/shoes.jpeg",
    answer: "been",
    level: 6
  },
  {
    questionId : 74,
    question: "Complete the following word '__ __ R R __ R' !",
    imageUrl: "images/mirror.jpeg",
    answer: "mirror",
    level: 6
  },
  {
    questionId : 75,
    question: "Complete the following word 'C __ __ T __ __ S' !",
    imageUrl: "images/clothes.jpeg",
    answer: "clothes",
    level: 6
  },
  {
    questionId : 76,
    question: "Complete the following word '__ O __ __ E' !",
    imageUrl: "images/house.jpeg",
    answer: "house",
    level: 6
  },
  {
    questionId : 77,
    question: "Complete the following word '__ __ H __ __ L !",
    imageUrl: "images/school.jpeg",
    answer: "school",
    level: 6
  },
  {
    questionId : 78,
    question: "Complete the following word '__ __ N __ __ L !",
    imageUrl: "images/pencil.jpeg",
    answer: "pencil",
    level: 6
  },
  {
    questionId : 79,
    question: "Complete the following word 'D __ __ __ H __ __ E !",
    imageUrl: "images/dolphine.jpeg",
    answer: "dolphine",
    level: 7
  },
  {
    questionId : 80,
    question: "Complete the following word 'A __ __ __ R __ __ M !",
    imageUrl: "images/aquarium.jpeg",
    answer: "aquarium",
    level: 7
  },
  {
    questionId : 81,
    question: "Complete the following word '__ __ __ S __ __ C __ M !",
    imageUrl: "images/capsicum.jpeg",
    answer: "capsicum",
    level: 7
  },
  {
    questionId : 82,
    question: "Complete the following word '__ __ __ H R __ __ M !",
    imageUrl: "images/mushroom.jpeg",
    answer: "mushroom",
    level: 7
  },
  {
    questionId : 83,
    question: "Complete the following word '__ __ N __ __ __ P __ E !",
    imageUrl: "images/pineapple.jpeg",
    answer: "pineapple",
    level: 7
  },
  {
    questionId : 84,
    question: "Complete the following word '__ __ T __ __ U __!",
    imageUrl: "images/octopus.jpeg",
    answer: "octopus",
    level: 7
  },

]
